src/server/game/DataStores/DBCStores.cpp

```diff
@@ -918,3 +919,50 @@ SkillRaceClassInfoEntry const* GetSkillRaceClassInfo(uint32 skill, uint8 race, u
 
     return NULL;
 }
+
+std::string GetDBCLocaleFolder(std::string const& dataPath)
+{
+    boost::filesystem::path result=dataPath;
+    boost::filesystem::directory_iterator end_iter;
+    std::string detectedLocalePath="";
+    int locale=ConfigMgr::instance()->GetIntDefault("DBC.Locale", 0);
+    for(boost::filesystem::directory_iterator dir_iter(dataPath); dir_iter != end_iter && detectedLocalePath.empty(); ++dir_iter)
+        if (boost::filesystem::is_directory(*dir_iter) && DBCLocaleFolderMatch(*dir_iter, locale))
+            detectedLocalePath=(*dir_iter).path().string(); //Get only the last part of the path, the locale part (esES, esMX, etc)
+
+    if (detectedLocalePath.empty())
+        TC_LOG_WARN("server.loading", "DBC files detected at DataDir/dbc root folder instead of its locale subfolder, this will be removed in a future");
```

Comment:
This doesn't mean that the files are in the root /dbc directory. The folder could be empty.
Also, removed in a future what?